### ISSUE TITLE
`-Wdangling-reference` reported in `llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#125538]
```
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp: In member function ‘bool {anonymous}::CombineRuleBuilder::emitCXXMatchApply(llvm::CodeExpansions&, llvm::gi::RuleMatcher&, llvm::ArrayRef<llvm::gi::CXXPattern*>)’:
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp:1910:15: warning: possibly dangling reference to a temporary [-Wdangling-reference]
 1910 |   const auto &Code = CXXPredicateCode::getCustomActionCode(CodeStr);
      |               ^~~~
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/GlobalISelCombinerEmitter.cpp:1910:59: note: the temporary was destroyed at the end of the full expression ‘llvm::gi::CXXPredicateCode::getCustomActionCode(std::__cxx11::basic_string<char>(CodeStr))’
 1910 |   const auto &Code = CXXPredicateCode::getCustomActionCode(CodeStr);
      |                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
```

I am using `gcc (GCC) 14.2.1 20240910` on Manjaro Linux.

**Comment by chrchr-github** [comment]
Looks like a false positive from GCC,
The returned const reference points into the `AllCXXCustomActionCode` parameter, which is a static object
~~~c++
static const CXXPredicateCode &getCustomActionCode(std::string Code) {
    return get(AllCXXCustomActionCode, std::move(Code));
  }
~~~

**Comment by firewave** [comment]
Well, it does say "warning: ***possibly*** dangling reference to a temporary"...
